### PR TITLE
[`pylint`]: Implement `continue-in-finally` (`E0116`)

### PR DESCRIPTION
**Pull request by latonis**
Rule reference: https://pycodequ.al/docs/pylint-messages/e0116-continue-in-finally.html

Implemented for Pylint parent issue: #970

### CODE REVIEW

File: crates/ruff/resources/test/fixtures/pylint/continue_in_finally.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        finally:
+            continue  # [continue-in-finally]
+        pass
+
+
+while True:
+    try:
+        pass
+    finally:
+        continue  # [continue-in-finally]
```
Comment (MichaReiser) [inline crates/ruff/resources/test/fixtures/pylint/continue_in_finally.py:24]: Can you add a test for nested statements?

```suggestion
    finally:
    	if True:
	        continue  # [continue-in-finally]
```

Comment (latonis) [reply crates/ruff/resources/test/fixtures/pylint/continue_in_finally.py:24]: yup, actually realized that right after pushing, better implementation incoming :)

**Comment by latonis** [comment]
I have fixed the nested catch @MichaReiser, apologies I didn't catch it initially 😸

File: crates/ruff/src/rules/pylint/rules/continue_in_finally.rs [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+            checker
+                .diagnostics
+                .push(Diagnostic::new(ContinueInFinally, Range::from(stmt)));
+        }
+
+        match &stmt.node {
+            StmtKind::If { body, .. }
+            | StmtKind::With { body, .. }
+            | StmtKind::AsyncWith { body, .. } => {
+                traverse_body(checker, body);
```
Comment (r3m0t) [inline crates/ruff/src/rules/pylint/rules/continue_in_finally.rs:57]: search for `Vec<Stmt<U>>` in `StmtKind` source found some other cases to traverse-

`orelse`  in a couple of nodes, including `If`, `For`, `AsyncFor` and `While`

Comment (latonis) [reply crates/ruff/src/rules/pylint/rules/continue_in_finally.rs:57]: I always forget those `for..else` and `while..else` statements exist 🫠

**Review by MichaReiser — APPROVED** [review]

**Pull request merged**